### PR TITLE
feat(013): conversation media bridge (send/receive attachments) [workspace#013]

### PR DESCRIPTION
**Pull request by antst**
Part of the cross-repo feature **`workspace#013-matrix-media-file-service`** — conversation media via file-service. This is the matrix-adapter slice.

## What this adds
Media/attachment support to the previously text-only Matrix bridge, while keeping the adapter **stateless** (no DB, no mapping table, no auth minting, no bucket logic).

- **Outbound (web→Element):** for each attachment, fetch bytes from file-service (`GET /internal/file/{id}/content`), `UploadBytes` → `mxc`, build an `m.image`/`m.video`/`m.audio`/`m.file` event carrying `url`, `info`, and a custom `io.alkemio.document_id` (so the server resolves our own echoes lookup-free). Text + N attachments → 1 `m.text` (if any) + N media events.
- **Inbound (Element→web):** branch on `msgtype`, extract `url`(→`media_id`), `info`, and `io.alkemio.document_id`, surfaced on the received-message payload for the server to re-home/resolve.
- **DTOs** regenerated into `@alkemio/matrix-adapter-lib` (`lib/src/dto/generated.ts`) — consumed by the server slice. *(The lib version bump/publish is a separate release step; the server pin updates after it publishes — see rollout ordering in the workspace plan.)*

## Design invariants
- Stateless: the `mxc ↔ document` link is carried by the file-service `externalReference` (inbound) + the `io.alkemio.document_id` event field (outbound) — no mapping table here.
- `info.size` uses the actual uploaded byte count, not caller-claimed size.
- Deterministic Matrix transaction IDs from an optional idempotency key make retries safe against duplicate events.

## Hardening from review (unbiased multi-lens pass + full remediation)
- **Concurrency:** `message.send` is now processed by a **room-partitioned bounded worker pool** — per-room ordering preserved, cross-room sends concurrent (`SEND_CONCURRENCY`, default 8), graceful drain; plus a per-send deadline (`SEND_TIMEOUT_SECONDS`, default 120). Fixes head-of-line blocking introduced by the longer media send path.
- **Trust boundary:** `io.alkemio.document_id` is only trusted from our own appservice-ghost senders (a forged field from an arbitrary in-room user is dropped → treated as Element-origin).
- **Read consistency:** `GetMessage` now returns attachments like the other read paths.
- **Render:** inbound media no longer duplicates the filename into message `Content`.
- Bounded/UUID-validated file-service fetch, boot-time config validation, dead code removed, shared msgtype consts, and broad test coverage (threaded media replies, video/audio, MIME fallback, multi-attachment, edge cases).

## Known limitation (documented)
A retried media send can orphan a Synapse `mxc` (Matrix upload is not idempotent; deterministic upload would require reintroducing state). Orphans are reclaimable by Synapse media retention.

## Tests / gates
`go build`, `go vet`, `go test ./...` (incl. `-race`), and `golangci-lint` all green.

🔗 Workspace spec: `agents-hq/specs/013-matrix-media-file-service/` · contract: `contracts/matrix-adapter.md` · ADR `docs/decisions/0001-matrix-media-storage-via-file-service.md`

## Summary by CodeRabbit

* **New Features**
  * Added media-attachment support for sending and receiving (images/video/audio/files) with document/media IDs, MIME type, size, display name, and optional dimensions.
  * Added `idempotency_key` support for safer retries with deterministic de-duplication.
  * Enabled per-room ordered outbound processing with bounded concurrent progress across rooms.
* **Bug Fixes**
  * Improved caption/body handling for MSC2530 vs legacy formats, including correct handling of explicitly empty captions and avoiding filename duplication.
  * Media parsing now rejects forged document identifiers and filters blank preview items while keeping requested roots.
  * Partial multi-send failures report the delivered primary event.
* **Documentation**
  * Updated media-attachment protocol and configuration docs, including `FILE_SERVICE_URL` and attachment limits.
* **Chores**
  * Added fail-fast configuration validation and stronger outbound attachment validation (count and required document id).

---
### 📦 PR image — test the current state of this PR

```bash
docker pull ghcr.io/alkem-io/matrix-adapter:pr-67
```

Current build: `ghcr.io/alkem-io/matrix-adapter:pr-67-2c70078` · `linux/amd64` + `linux/arm64` · index digest `sha256:7b290ef0c7b4fd4e5fa3c4e38c77bf6dbc9af274ff28f25c52452fbb1bd561b4`

### CODE REVIEW

File: internal/infrastructure/matrix/listener.go [hunk 1]
```diff
@@ -801,6 +823,138 @@ func (m *MautrixAdapter) isSpaceRoom(ctx context.Context, roomID id.RoomID) (boo
...
+// appservice controls on our homeserver: a UUID-localpart ghost (or the bot,
+// which is itself a UUID localpart) on the configured homeserver domain. This
+// is the trust gate for the io.alkemio.document_id breadcrumb — only echoes
+// from our ghosts may surface an authoritative DocumentID.
+func (m *MautrixAdapter) isOwnAppserviceUser(userID id.UserID) bool {
+	if userID.Homeserver() != m.idMapper.HomeserverDomain() {
+		return false
+	}
+	return m.idMapper.AlkemioActorID(userID) != uuid.Nil
+}
```
[the remark this answers is not in the document]
Comment (antst) [reply internal/infrastructure/matrix/listener.go]: Resolving as moot — the code this anchors to no longer exists, and we went further than the suggestion.

`isOwnAppserviceUser` has been **deleted entirely** rather than tightened to an exclusive namespace. The gate could not work as conceived: the adapter impersonates actor X *as* X, so its ghost MXID is byte-identical to that actor's own OIDC login — the appservice namespace `@[0-9a-fA-F-]{36}:.*` is deliberately non-exclusive. Any sender-shape test therefore reduces to "the localpart parses as a UUID", which is true for every real user on the homeserver. It could not distinguish anything.

`io.alkemio.document_id` is now documented as an **unauthenticated hint**. Its authority comes from the server-side `bucket + createdBy == sender` check, not from the breadcrumb — see the comment at `listener.go:908-931`.

Also worth noting for future reviews: this homeserver is not federated and never will be, so there is no remote attacker. The realistic threat is a local room member, and that is exactly what the server-side gate covers.

File: internal/infrastructure/queue/worker_pool.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		}
+		msg := backlog[0]
+		p.queues[key] = backlog[1:]
+		p.mu.Unlock()
+
+		p.sem <- struct{}{} // bound concurrent handlers across all partitions
+		p.process(msg)
+		<-p.sem
+	}
+}
```
[the remark this answers is not in the document]
Comment (antst) [reply internal/infrastructure/queue/worker_pool.go]: Resolving as moot — `internal/infrastructure/queue/worker_pool.go` has been **deleted**, so the panic-recovery concern has no subject.

It went with the rest of the send-path hardening layer (deterministic per-event transaction ids, partial-send markers, room-partitioned ordered dispatch, and the aggregate attachment memory budget), which was built on this branch and then deliberately stripped back to the develop baseline in `4263ce7`.

That was a product decision, not an oversight: a multi-attachment send is 1 + N independent Matrix events and is **not** atomic — exactly as in Element. The machinery duplicated, imperfectly, semantics that belong to Synapse and to the sending client, and each round of fixes traded one subtle failure mode for another. `attachmentBudget` / `MaxTotalAttachmentBytes` / `acquireAttachmentBudget` now return zero hits repo-wide; the only remaining cap is the per-attachment `FILE_SERVICE_MAX_ATTACHMENT_BYTES`.
